### PR TITLE
feature: Toast component

### PR DESCRIPTION
**Pull request by hervedombya**
**Component**:
'Toast'

**Description**:
Toast inform users of a process that an app has performed or will perform. They appear temporarily, towards the top of the screen. They shouldn't interrupt the user experience, and they don't require user input to disappear.

**Design**:
Enregistrement de l’écran 2023-09-18 à 16 36 07

**Breaking Changes**:

- [] Breaking Changes

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: src/lib/components/toast/Toast.component.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    default:
+      return 'Info-circle';
+  }
+};
+
+const DefaultIcon = ({ status }: { status: string }) => {
+  const color = useGetBackgroundColor(status);
+  const iconName = defaultIconName(status);
+
+  return <Icon name={iconName} color={color as IconColor} />;
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/Toast.component.tsx]: Why do we need to cast the IconColor type here ? Is this type not accurate ? maybe we can update it if so 🤔

File: src/lib/components/toast/Toast.component.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    default:
+      return theme.infoPrimary;
+  }
+};
+
+const useGetRgbBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return 'rgba(10, 173, 166, 0.4)';
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/Toast.component.tsx]: nit: I wonder if those rgba shouldn't either be added to the theme either be derived from one of the theme variable

File: src/lib/components/toast/ToastPositionHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  'top-right': {
+    top: '3rem',
+    right: '1rem',
+  },
+  'top-center': {
+    top: '1rem',
+    left: '50%',
+    transform: 'translateX(-50%)',
+  },
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/ToastPositionHelpers.ts:24]: Does those default 'top' position already take the navbar in account ?

File: src/lib/components/toast/ToastPositionHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+export type ToastPosition =
+  | 'top-left'
+  | 'top-right'
+  | 'top-center'
+  | 'bottom-left'
+  | 'bottom-right'
+  | 'bottom-center';
+
+export const positionOutput: Record<ToastPosition, {}> = {
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/ToastPositionHelpers.ts]:
```suggestion
export const positionOutput: Record<ToastPosition, CSSProperties> = {
```

File: src/lib/components/toast/ToastProgressBar.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { darken } from 'polished';
+import { useEffect, useState } from 'react';
+import { ProgressBar } from '../progressbar/ProgressBar.component';
+
+export function ToastProgressBar({
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/ToastProgressBar.tsx]: nit: I wonder wether we shouldn't enlarge the name of this component. It's more like a 'fake' progress bar based on a duration than a Toast specific progress bar
```suggestion
export function DurationBasedProgressBar({
```

File: src/lib/components/toast/useMutationsHandler.test.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+      mutation: {
+        isLoading: false,
+        isSuccess: false,
+        isError: false,
+        isIdle: true,
+      },
+      name: 'mutation3',
+      isPrimary: false,
+    },
+  ] as MutationConfig<unknown, unknown>[];
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/useMutationsHandler.test.tsx]: I think the generics should remain on the mutation level, we can have several mutation in this array and each of them with their own variable and return type

File: src/lib/components/toast/Toast.component.tsx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+  if (!open) {
+    return null;
+  }
+
+  return (
+    <div
+      ref={ref}
+      role="alert"
+      aria-live="polite"
```
Comment (JBWatenbergScality) [inline src/lib/components/toast/Toast.component.tsx]: 👍 does the accessibility spec on toast mention the fact that we need to add a signaler for errors ? something like aria-invalid, but probably not that one as it is reserved for fields iirc

File: src/lib/components/toast/ToastPositionHelpers.ts (hunk 3 above)
Comment (hervedombya) [reply src/lib/components/toast/ToastPositionHelpers.ts:24]: yes

File: src/lib/components/toast/Toast.component.tsx (hunk 1 above)
Comment (hervedombya) [reply src/lib/components/toast/Toast.component.tsx]: `IconColor` type is `keyof typeof theme` but actually Typescript considers a variable `theme.statusHealthy` as a string

File: src/lib/components/toast/Toast.component.tsx (hunk 2 above)
Comment (hervedombya) [reply src/lib/components/toast/Toast.component.tsx]: opacity can vary from case to case in my opinion, but the rgb value doesn't, so maybe just add the RGB value?

File: src/lib/components/toast/Toast.component.tsx (hunk 7 above)
Comment (hervedombya) [reply src/lib/components/toast/Toast.component.tsx]: https://docs.google.com/document/d/12HoCYmkDvDMATOOJWdryoSwghHFoisA9OQdiPqCqgvY/edit#heading=h.yx1g2yktvxqs not in documentation, you know where is mentionned ?

File: src/lib/components/toast/Toast.component.tsx (hunk 2 above)
Comment (JBWatenbergScality) [reply src/lib/components/toast/Toast.component.tsx]: Agree !

**Review by JBWatenbergScality — APPROVED** [review]

**Pull request merged**